### ISSUE TITLE
"ts.check" function return values different from the documents

### ISSUE THREAD

**Issue by marsCatXdu** [rpm-software-management/rpm#1871]
I'm currently working on a GUI rpm software manager, it uses rpm python APIs to work. And I read the 16th chapter in 'fedora rpm guide'  as the api document.
When I try to check package's( I know this package lacks the dependencies, doing this for a test ) dependencies with ts.check() function,  the function returns tuples different from the document.
For example, I'm using fedora 34 and I tried to install 'qtbase 5.9' package from centos7, the function returns:
```
[(('qt5-qtbase', '5.9.7', '4.el7'), ('libcrypto.so.10()(64bit)', ''), 0, 0, None), 
(('qt5-qtbase', '5.9.7', '4.el7'), ('libcrypto.so.10(OPENSSL_1.0.1_EC)(64bit)', ''), 0, 0, None), 
(('qt5-qtbase', '5.9.7', '4.el7'), ('libcrypto.so.10(OPENSSL_1.0.2)(64bit)', ''), 0, 0, None) .....]
```
The 2nd, 3rd value in tuple seems like enums, and the 4th value is something else I don't know. The document says it supposed to be like:
`((N,V,R), (reqN, reqV), needsFlags, suggestedPackage, sense)`
I don't think they are the same thing, I guess there maybe something wrong in order.

And, the doc says "You can tell whether the dependency is a conflict or a requirement based on the sense value, one of
rpm.RPMSENSE_CONFLICTS or rpm.RPMSENSE_REQUIRES."  I didn't find the mentioned enum RPMSENSE_CONFLICTS and RPMSENSE_REQUIRES, and they are not exist in present rpm code, but they still exists in [this comment](https://github.com/rpm-software-management/rpm/blob/master/python/rpmts-py.c#L91)

Thanks for help first.

**Comment by pmatilai** [comment]
It's not rpm.RPMSENSE_CONFLICTS, never was, but rpm.RPMDEP_SENSE_CONFLICTS (and REQUIRES). Those still exist as per the docs.

And yeah it seems the order got messed up in commit fab2debfe440d677dbd072c3cd73d2c99876e7a5, which means that apparently nobody has used this in the last 12+ years. I'm half surprised the callback works at all anymore...

**Comment by pmatilai** [comment]
The order should be fixed now as of commit f0c158cbc8a50a776b44de2c0fe744c451155a41, thanks for reporting. 

But considering this state of things... I would be surprised if there were no other bugs and gotchas in this direction. For one, rich dependencies didn't exist when this mechanism was last widely used, and ditto for various other things. These days, everybody seems to be using libsolv as that's the only thing that can actually sort out rich dependencies and all.
